### PR TITLE
f-2: Adding MultiThreading

### PR DESCRIPTION
**Pull request by ValeryEMalakhov**

### CODE REVIEW

File: src/main/scala/com/prb/f1/spiralStrategies/MatrixSpiralPartBuilder.scala [hunk 1]
```diff
@@ -0,0 +1,43 @@
+package com.prb.f1.spiralStrategies
+
+/**
+  * Contains a <code>run</code> method
+  * for matrix iteration part building.
+  */
+class MatrixSpiralPartBuilder(n: Int,
+                              iterationFirstValue: Int,
```
Comment (btrofimov) [inline src/main/scala/com/prb/f1/spiralStrategies/MatrixSpiralPartBuilder.scala]: scala style requires diffrerent identation

File: src/main/scala/com/prb/f1/spiralStrategies/MultiThreadedMatrixSpiralBuilderImpl.scala [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    if (n < 2) throw TooSmallNException()
+
+    val fillableMatrix: Array[Array[Int]] = Array.ofDim[Int](n, n)
+
+    var currentN = n
+    var iterationFirstValue = 1
+    var iterationLastValue = 1
+
+    //  val commonPool: ForkJoinPool = new ForkJoinPool((N.toDouble / 2).ceil.toInt)
+    val iterationExecutor: ExecutorService = Executors.newFixedThreadPool(Runtime.getRuntime.availableProcessors())
```
Comment (btrofimov) [inline src/main/scala/com/prb/f1/spiralStrategies/MultiThreadedMatrixSpiralBuilderImpl.scala]: iterationExecutor should be passed as parameter

File: src/main/scala/com/prb/f1/spiralStrategies/MultiThreadedMatrixSpiralBuilderImpl.scala [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+    /*
+    * tried to use `ArrayBuffer`, but util.concurrent.ExecutorService
+    * work only with java.util collections, such as `ArrayList`
+    * */
+    val iterationList = new util.ArrayList[Callable[AnyRef]]((n.toDouble / 2).ceil.toInt)
+
+    for (iterationStartingIndex <- 0 until (n.toDouble / 2).ceil.toInt) {
+
+      iterationLastValue += f(n - (iterationStartingIndex * 2))
```
Comment (btrofimov) [inline src/main/scala/com/prb/f1/spiralStrategies/MultiThreadedMatrixSpiralBuilderImpl.scala]: please define *def f* right here

File: src/main/scala/com/prb/f1/spiralStrategies/MultiThreadedMatrixSpiralBuilderImpl.scala [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+    // Calculates the required values for each spiral of the matrix
+    // and passes them to the threads.
+    for (iterationStartingIndex <- 0 until (n.toDouble / 2).ceil.toInt) {
+
+      iterationLastValue += calculateNextIterationValue(n - iterationStartingIndex * 2)
+
+      if (iterationLastValue == iterationFirstValue) iterationLastValue += 1
+
+      iterationExecutor.execute(
```
Comment (ValeryEMalakhov) [inline src/main/scala/com/prb/f1/spiralStrategies/MultiThreadedMatrixSpiralBuilderImpl.scala]: MatrixSpiralBuilderFactory.createExecutorService.execute( //... ?

**Pull request merged**